src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -384,19 +371,15 @@ private int implWrite(ByteBuffer src) throws IOException {
     }
 
     private int traceImplWrite(ByteBuffer src) throws IOException {
-        int bytesWritten = 0;
+        int bytes = 0;
         long start = 0;
         try {
             start = FileWriteEvent.timestamp();
-            bytesWritten = implWrite(src);
+            bytes = implWrite(src);
         } finally {
-            long duration = FileWriteEvent.timestamp() - start;
-            if (FileWriteEvent.shouldCommit(duration)) {
-                long bytes = bytesWritten > 0 ? bytesWritten : 0;
-                FileWriteEvent.commit(start, duration, path, bytes);
-            }
+            FileWriteEvent.offer(start, path, bytes);
         }
-        return bytesWritten;
+        return bytes;
```

Comment:
The method returns the number of bytes written and traceImplRead uses bytesRead for the number of bytes read. So I think better to leave it as bytesWritten.